readme.md

```diff
@@ -92,36 +155,59 @@ The `pkcs1-reader-sun` does not include any extra dependencies, but assumes the
 available as part of most popular JVM implementations, including the Oracle and OpenJDK JVMs for
 Java 7 and Java 8.
 
-### Add Dependency Using Gradle
+## error-handling
+Provides utilities for relaying Java exceptions across JVM boundaries by serializing exceptions as JSON POJOs.
 
-```groovy
-repositories {
-  jcenter()
-}
-dependencies {
-  compile "com.palantir.remoting:http-clients:$version"
-  compile "com.palantir.remoting:retrofit-clients:$version"
-  compile "com.palantir.remoting:http-servers:$version"
-  compile "com.palantir.remoting:ssl-config:$version"
-  compile "com.palantir.remoting:error-handling:$version"
-  // support for PEM key store type using Bouncy Castle libraries
-  // compile "com.palantir.remoting:pkcs1-reader-bouncy-castle:$version"
-  // support for PEM key store type using Sun libraries
-  // compile "com.palantir.remoting:pkcs1-reader-sun:$version"
-}
-```
 
+# API Contract
+
+http-remoting makes the following opinionated customizations to the standard Dropwizard/Feign/Retrofit behavior.
+
+#### Object serialization/deserialization
+
+All parameters and return values of `application/json` endpoints are serialized/deserialized to/from JSON using a
+Jackson `ObjectMapper` with `GuavaModule` and `Jdk7Module`. Servers must not expose parameters or return values that
+cannot be handled by this object mapper.
+
+#### Error propagation
```

Comment:
I think we should also define the `SerializableError` format as a specification here so that non-Java servers and non-Java clients know how to implement this.